index.js

```diff
@@ -210,9 +242,16 @@ async function underPressure (fastify, opts) {
         throw underPressureError
       }
     }
+    if (req.isUnderPressureError) {
+      return { status: 'not-ok' }
+    }
     return { status: 'ok' }
   }
 
+  async function onStatusDetails () {
```

Comment:
Why wrap a sync function in a promise?

```suggestion
  function onStatusDetails () {
```